static/js/util/profile_edit.js

```diff
@@ -440,13 +410,10 @@ export function boundDateField(keySet, label, omitDay) {
 
 /**
  * Validates the profile then PATCHes the profile if validation succeeded.
- *
- * @param validator {func} The function used to validate profile
- * @param isLastStep {bool} If true, this is the last tab in the profile
- * @returns {Promise} A promise which resolves with the new profile if validation and PATCH succeeded,
+ * Returns a promise which resolves with the new profile if validation and PATCH succeeded,
  * or rejects if either failed
  */
-export function saveProfileStep(validator, isLastStep=false) {
+export function saveProfileStep(validator: Function, isLastStep: boolean = false): Promise {
```

Comment:
Also can you use `Promise<Profile>` here? It looks like it handles the resolve case but for the reject case it will just return `any`, so we can't specify a type for that here